### PR TITLE
Add warning when submitting unchanged ranking

### PR DESCRIPTION
**Pull request by stephancill**
Fix for #272.

When the list is unchanged and the user tries to submit, they are presented with this modal:
Screenshot 2023-01-04 at 01 19 43

Something that may be an issue is that once the modal is presented, if the user presses Cancel, and they click submit again, it won't be presented again. More information on L27 of website/src/components/Survey/TaskControlsOverridable.tsx. Not sure how important this is but will required additional refactoring to fix.

### CODE REVIEW

File: website/src/pages/evaluate/rank_assistant_replies.tsx
Comment (fozziethebeat) [inline website/src/pages/evaluate/rank_assistant_replies.tsx:30]: Could we drop this?

**Review by fozziethebeat — CHANGES_REQUESTED** [review]
Looks good with one small change needed.  Will approve after that and after verifying another PR didn't cause a merge conflict.

File: website/src/pages/evaluate/rank_assistant_replies.tsx
Comment (stephancill) [reply website/src/pages/evaluate/rank_assistant_replies.tsx]: Oops, my bad. Fixed.

**Comment by andreaskoepf** [comment]
(I removed the backend label since the proposed change is in the website code.)

**Review by fozziethebeat — APPROVED** [review]

**Pull request merged**
